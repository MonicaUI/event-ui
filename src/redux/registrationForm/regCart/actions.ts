import {
  REMOVE_INVALID_CUSTOM_FIELD,
  RESET_REG_CART_DISABLE_REG,
  SET_REG_CART_FIELD_VALUE,
  SET_TEMPORARY_GUEST_FIELD_VALUE
} from './actionTypes';

function setRegCartFieldValue(path, value) {
  return {
    type: SET_REG_CART_FIELD_VALUE,
    payload: { path, value }
  };
}

function eventRegistrationPrefix(eventRegistrationId) {
  return ['eventRegistrations', eventRegistrationId];
}

export function setEventRegistrationFieldValue(
  eventRegistrationId: $TSFixMe,
  path: $TSFixMe,
  value: $TSFixMe
): $TSFixMe {
  return setRegCartFieldValue([...eventRegistrationPrefix(eventRegistrationId), ...path], value);
}

export function setTemporaryGuestFieldValue(path: $TSFixMe, value: $TSFixMe): $TSFixMe {
  return {
    type: SET_TEMPORARY_GUEST_FIELD_VALUE,
    payload: { path, value }
  };
}

export function removeInvalidCustomField(eventRegistrationId: $TSFixMe, fieldId: $TSFixMe): $TSFixMe {
  return {
    type: REMOVE_INVALID_CUSTOM_FIELD,
    payload: {
      fieldId,
      path: [...eventRegistrationPrefix(eventRegistrationId), 'attendee', 'personalInformation', 'customFields']
    }
  };
}

export function updateAdmin(admin: $TSFixMe) {
  return async (dispatch: $TSFixMe): Promise<$TSFixMe> => {
    dispatch(setRegCartFieldValue(['admin'], admin));
  };
}

export function updateEventRegistrations(eventRegistrations: $TSFixMe) {
  return async (dispatch: $TSFixMe): Promise<$TSFixMe> => {
    dispatch(setRegCartFieldValue(['eventRegistrations'], eventRegistrations));
  };
}

export function updateEventVoucher(eventRegistrationId: $TSFixMe, voucherCode: $TSFixMe) {
  return async (dispatch: $TSFixMe): Promise<$TSFixMe> => {
    dispatch(
      setRegCartFieldValue(
        [...eventRegistrationPrefix(eventRegistrationId), 'appliedVoucher', 'voucherCode'],
        voucherCode
      )
    );
  };
}

export function setAppointmentsAvailability(eventRegistrationId: $TSFixMe, availability: $TSFixMe): $TSFixMe {
  return setRegCartFieldValue(
    [...eventRegistrationPrefix(eventRegistrationId), 'appointmentsAvailability'],
    availability
  );
}

export function setAppointmentsMeetingInterest(eventRegistrationId: $TSFixMe, meetingInterest: $TSFixMe): $TSFixMe {
  return setRegCartFieldValue(
    [...eventRegistrationPrefix(eventRegistrationId), 'appointmentsMeetingInterest'],
    meetingInterest
  );
}

export function setSendEmailFlag(sendEmail: $TSFixMe): $TSFixMe {
  return setRegCartFieldValue(['sendEmail'], sendEmail);
}

export function saveLocaleId(localeId: $TSFixMe): $TSFixMe {
  return setRegCartFieldValue(['localeId'], localeId);
}

/**
 * Creates a redux action that replaces the regcart object in the redux state with the passed regcart
 * and sets a flag in the redux state that tells the code to prevent registration actions
 *
 * For now, this use case comes into picture in container events only
 * When a user can't access website pages but can access registration pages
 * and they land on a registration page in such an event
 * and reg cart creation is prevented due to the status/config of the event (like registration being closed etc)
 * then we keep the user on the registration page (because they dont have access to website pages)
 * and disable registration actions because we detected earlier that reg cart creation is not possible
 */
export const disableRegistration = (regPathId: $TSFixMe): $TSFixMe => {
  return {
    type: RESET_REG_CART_DISABLE_REG,
    payload: {
      regCart: {
        regCartId: '',
        eventRegistrations: {
          '00000000-0000-0000-0000-000000000001': {
            eventRegistrationId: '00000000-0000-0000-0000-000000000001',
            registrationPathId: regPathId,
            sessionRegistrations: {},
            attendee: {
              personalInformation: {
                firstName: '',
                lastName: ''
              }
            }
          }
        }
      }
    }
  };
};
