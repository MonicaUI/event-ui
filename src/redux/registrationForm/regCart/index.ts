import { lazyLoadAction, lazyLoadFunction } from '../../../utils/lazyLoad';

export const updateSessionRegistration = lazyLoadFunction(() =>
  import(/* webpackChunkName: "registration" */ './internal').then(m => m.updateSessionRegistration)
);

export const updateAdmissionItemRegistration = lazyLoadFunction(() =>
  import(/* webpackChunkName: "registration" */ './internal').then(m => m.updateAdmissionItemRegistration)
);

export const getPricing = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './pricing').then(m => (m as $TSFixMe).getPricing)
);

export const updateRegCartWithGuests = lazyLoadFunction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.updateRegCartWithGuests)
);

export const updateGuestsInRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.updateGuestsInRegCart)
);

export const setCurrentGuestEventRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.setCurrentGuestEventRegistration)
);

export const removeGuestByEventRegistrationId = lazyLoadFunction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.removeGuestByEventRegistrationId)
);

export const updateGuestDetails = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.updateGuestDetails)
);

export const clearTemporaryGuestInformation = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './guests').then(m => m.clearTemporaryGuestInformation)
);

export const removeEventRegistrationFromRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './group').then(m => m.removeEventRegistrationFromRegCart)
);

export const removeGroupMembersFromRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './group').then(m => m.removeGroupMembersFromRegCart)
);

export const addGroupMemberInRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './group').then(m => m.addGroupMemberInRegCart)
);

export const navigateToGroupMemberRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './group').then(m => m.navigateToGroupMemberRegistration)
);

export const startRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startRegistration)
);

export const abortRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.abortRegCart)
);

export const abortRegCartAndLogout = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.abortRegCartAndLogout)
);

export const rescindAbortRegCartAndLogoutRequest = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.rescindAbortRegCartAndLogoutRequest)
);

export const startModification = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startModification)
);

export const finalizeRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.finalizeRegistration)
);

export const calculateServiceFeesForPartialPayments = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.calculateServiceFeesForPartialPayments)
);

export const resumeAlreadyStartedCheckout = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.resumeAlreadyStartedCheckout)
);

export const waitForRegCartCheckoutCompletionUi = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.waitForRegCartCheckoutCompletionUi)
);

export const startCancelRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startCancelRegistration)
);

export const finalizeCancelRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.finalizeCancelRegistration)
);

export const startDeclineRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startDeclineRegistration)
);

export const startWaitlistRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.startWaitlistRegistration)
);

export const finalizeWaitlistRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.finalizeWaitlistRegistration)
);

export const finalizeDeclineRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.finalizeDeclineRegistration)
);

export const saveRegistration = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.saveRegistration),
  true
);

export const updateRegCart = lazyLoadFunction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.updateRegCart)
);

export const updatePaymentCreditsInRegCart = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './paymentCredits').then(m => m.updatePaymentCreditsInRegCart)
);

export const updateTravelFlagInRegCart = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.updateTravelFlagInRegCart)
);

export const setAirRequestOptOutChoice = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './workflow').then(m => m.setAirRequestOptOutChoice)
);

export const applyPartialEventRegistrationUpdate = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () =>
    import(/* webpackChunkName: "registration" */ './partialUpdates').then(m => m.applyPartialEventRegistrationUpdate)
);

export const setRegistrationTypeId = lazyLoadAction(
  // eslint-disable-next-line import/no-cycle
  () => import(/* webpackChunkName: "registration" */ './registrationTypes').then(m => m.setRegistrationTypeId)
);

export const runRegistrationTypeChangeValidationsForPrimaryAndGuest = lazyLoadFunction(
  // eslint-disable-next-line import/no-cycle
  () =>
    import(/* webpackChunkName: "registration" */ './registrationTypes').then(
      m => m.runRegistrationTypeChangeValidationsForPrimaryAndGuest
    )
);

export const restoreRegistration = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './restore').then(m => m.restoreRegistration)
);

export const restoreRegistrationFromOtherTab = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './restore').then(m => m.restoreRegistrationFromOtherTab)
);

export const resumePartialRegistration = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './restore').then(m => m.resumePartialRegistration)
);

export const searchPartialRegistration = lazyLoadAction(() =>
  import(/* webpackChunkName: "registration" */ './restore').then(m => m.searchPartialRegistration)
);
