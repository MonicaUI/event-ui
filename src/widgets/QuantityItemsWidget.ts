import { connect } from 'react-redux';
import QuantityItemsWidget from 'event-widgets/lib/QuantityItems/QuantityItemsWidget';
import { getEventTimezone } from '../redux/reducer';
import {
  getPrimarySortedVisibleQuantityItems,
  getEventRegistrationForPrimaryRegistrant,
  getQuantityItemInfoForPrimary
} from '../redux/selectors/productSelectors';
import { withLoading } from 'nucleus-guestside-site/src/redux/modules/dialogContainer';
import { updateQuantity, updateLocalQuantity } from '../redux/registrationForm/regCart/quantityItems';
import { getAdvancedQuantityItemRules } from '../redux/selectors/event';
import {
  getEventRegistrationId,
  isRegistrationModification,
  modificationStart,
  getRegistrationTypeId,
  getSelectedAdmissionItemDefinition,
  isRegApprovalRequired
} from '../redux/selectors/currentRegistrant';
import { SAVING_REGISTRATION } from '../redux/registrationIntents';
import {
  optionalItemIsVisibleForRegistrationType,
  optionalItemIsVisibleForAdmissionItem,
  areRegistrationActionsDisabled
} from '../redux/selectors/shared';

/**
 * Data wrapper for the QuantityItems widget.
 */
export default connect(
  (state: $TSFixMe) => {
    const {
      text: {
        resolver: { currency }
      },
      capacity
    } = state;
    const { getRegisteredQuantityItems } = modificationStart;
    const quantityItems = getPrimarySortedVisibleQuantityItems(state);
    const eventRegistrationId = getEventRegistrationId(state);
    const registeredQuantityItems = getRegisteredQuantityItems(state);
    const isSelectDisabled =
      (state.regCartStatus && state.regCartStatus.registrationIntent === SAVING_REGISTRATION) ||
      areRegistrationActionsDisabled(state);

    const selectedAdmissionItem = getSelectedAdmissionItemDefinition(state);
    const registrationTypeId = getRegistrationTypeId(state);
    const quantityItemsVisibleToPrimary =
      quantityItems &&
      Object.values(quantityItems)
        .filter(quantityItem => {
          return (
            optionalItemIsVisibleForRegistrationType(registrationTypeId, quantityItem) &&
            optionalItemIsVisibleForAdmissionItem(selectedAdmissionItem, quantityItem)
          );
        })
        .map(quantityItem => (quantityItem as $TSFixMe).id);
    return {
      eventRegistrationId,
      quantityItems,
      capacity,
      currency,
      eventTimezone: getEventTimezone(state),
      currentPrimaryRegistrant: getEventRegistrationForPrimaryRegistrant(state),
      defaultQuantityLimit: 32000,
      advancedQuantityItemRules: getAdvancedQuantityItemRules(state),
      primaryQuantityItemInfo: getQuantityItemInfoForPrimary(state),
      overrideFullCapacity: state.defaultUserSession.isPlanner,
      isRegMod: isRegistrationModification(state),
      registeredQuantityItems,
      isSelectDisabled,
      visibleToPrimary: quantityItemsVisibleToPrimary,
      isRegApprovalRequired: isRegApprovalRequired(state),
      isPreview: state.defaultUserSession.isPreview
    };
  },
  {
    updateQuantity: withLoading(updateQuantity),
    updateLocalQuantity: withLoading(updateLocalQuantity)
  },
  (stateProps: $TSFixMe, dispatchProps: $TSFixMe, ownProps: $TSFixMe) => {
    return {
      ...ownProps,
      ...stateProps,
      ...dispatchProps,
      updateQuantity: dispatchProps.updateQuantity.bind(null, stateProps.eventRegistrationId),
      updateLocalQuantity: dispatchProps.updateLocalQuantity.bind(null, stateProps.eventRegistrationId)
    };
  }
  // @ts-expect-error ts-migrate(2345) FIXME: Argument of type 'typeof QuantityItemsWidget' is n... Remove this comment to see the full error message
)(QuantityItemsWidget);
