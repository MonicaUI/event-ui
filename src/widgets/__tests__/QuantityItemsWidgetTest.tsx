import React from 'react';
import renderer from 'react-test-renderer';
import QuantityItemsWidget from '../QuantityItemsWidget';
import { mount } from 'enzyme';
import { wait } from '../../testUtils';
import { updateQuantity, updateLocalQuantity } from '../../redux/registrationForm/regCart/quantityItems';

jest.mock('../../redux/registrationForm/regCart/quantityItems', () => {
  return {
    updateQuantity: jest.fn(),
    updateLocalQuantity: jest.fn()
  };
});

function getState() {
  return {
    event: {
      timezone: 35,
      products: {
        quantityItems: {
          quantityItem1: {
            id: 'quantityItem1',
            capacityId: 'quantityItem1'
          },
          quantityItemNoFee: {
            id: 'quantityItemNoFee',
            capacityId: 'quantityItemNoFee'
          },
          quantityItem2WithCap1: {
            id: 'quantityItem2WithCap1',
            capacityId: 'quantityItem2WithCap1'
          }
        }
      }
    },
    website: {
      pluginData: {
        registrationProcessNavigation: {
          registrationPaths: {
            regPathId: {
              pageIds: ['regProcessStep1', 'regProcessStep2']
            }
          }
        }
      }
    },
    appData: {
      registrationSettings: {
        registrationPaths: {
          regPathId: {}
        }
      }
    },
    pathInfo: {
      currentPageId: 'regProcessStep2'
    },
    registrationForm: {
      currentEventRegistrationId: 'eventRegistrationId',
      regCart: {
        regCartId: 'regCartId',
        eventRegistrations: {
          eventRegistrationId: 'eventRegistrationId'
        }
      }
    },
    capacity: {
      quantityItem1: {
        active: true,
        availableCapacity: 123
      },
      quantityItem2WithMaxPerInvitee: {
        active: true,
        availableCapacity: 1
      }
    },
    text: {
      resolver: {
        date: () => 'some date ',
        currency: x => x
      },
      translate: x => x
    },
    timezones: {
      35: {
        id: 35,
        name: 'Eastern Time',
        nameResourceKey: 'Event_Timezone_Name_35__resx',
        plannerDisplayName: '(GMT-05:00) Eastern [US & Canada]',
        hasDst: true,
        utcOffset: -300,
        abbreviation: 'ET',
        abbreviationResourceKey: 'Event_Timezone_Abbr_35__resx',
        dstInfo: []
      }
    },
    defaultUserSession: {
      isPlanner: false
    },
    visibleProducts: {
      Sessions: {
        eventRegistrationId: {
          quantityItems: {
            quantityItem1: {
              id: 'quantityItem1',
              capacityId: 'quantityItem1',
              name: 'Quantity Item Name',
              code: 'Quantity Item Code',
              description: 'Quantity Item Description',
              defaultFeeId: 'fee1',
              fees: {
                fee1: {
                  amount: 100,
                  name: 'Fee 1',
                  registrationTypes: [],
                  chargePolicies: [
                    {
                      amount: 100,
                      effectiveUntil: '2999-12-31T00:00:00.000Z',
                      isActive: true
                    },
                    {
                      amount: 50,
                      effectiveUntil: new Date(new Date().valueOf() + 1000 * 3600 * 24 * 30).toISOString(),
                      isActive: true
                    }
                  ]
                }
              },
              displayOrder: 1,
              isOpenForRegistration: true,
              associatedRegistrationTypes: [],
              status: 2
            },
            quantityItemNoFee: {
              id: 'quantityItemNoFee',
              capacityId: 'quantityItem1',
              name: 'Quantity Item No Fee',
              code: 'quantity Item Code 2',
              description: 'Quantity Item Description',
              defaultFeeId: '00000000-0000-0000-0000-000000000000',
              fees: {},
              displayOrder: 2,
              isOpenForRegistration: true,
              status: 2,
              associatedRegistrationTypes: []
            },
            quantityItem2WithMaxPerInvitee: {
              id: 'quantityItem2WithMaxPerInvitee',
              capacityId: 'quantityItem2WithMaxPerInvitee',
              name: 'Quantity Item 2 With Cap 1',
              code: 'Quantity Item 2 Code',
              description: 'Quantity Item 2 Description',
              defaultFeeId: 'fee1',
              fees: {
                fee1: {
                  amount: 100,
                  name: 'Fee 1',
                  registrationTypes: [],
                  chargePolicies: [
                    {
                      amount: 100,
                      effectiveUntil: '2999-12-31T00:00:00.000Z',
                      isActive: true
                    },
                    {
                      amount: 50,
                      effectiveUntil: new Date(new Date().valueOf() + 1000 * 3600 * 24 * 30).toISOString(),
                      isActive: true
                    }
                  ]
                }
              },
              status: 2,
              displayOrder: 3,
              isOpenForRegistration: true,
              associatedRegistrationTypes: [],
              quantity: 2
            }
          }
        }
      }
    }
  };
}

async function dispatch(action) {
  if (typeof action === 'function') {
    await action(dispatch, getState);
  }
}

const subscribe = () => {};

const defaultProps = {
  classes: {},
  style: {},
  translate: (res, opts) => (opts ? `${res}:${JSON.stringify(opts)}` : res),
  config: {
    headerText: 'Header',
    instructionalText: 'Instructional text',
    display: {
      capacity: true,
      description: true,
      fees: true,
      itemCode: true
    }
  },
  isRegistrationPage: true,
  'data-cvent-id': 'widget-QuantityItems-widget:quantityItems',
  store: { dispatch, getState, subscribe }
};

describe('QuantityItemsTest', () => {
  test('should render', () => {
    const component = renderer.create(<QuantityItemsWidget {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });

  test('should select', async () => {
    const wrapper = mount(<QuantityItemsWidget {...defaultProps} />);
    wrapper
      .find('[data-cvent-id="quantityItems-item-quantityItem1"]')
      .find('[data-cvent-id="input"]')
      .hostNodes()
      .simulate('change', { target: { value: '2' } });
    await wait(0);
    expect(updateQuantity).toHaveBeenCalled();
  });

  test('should add to local state', async () => {
    const wrapper = mount(<QuantityItemsWidget {...defaultProps} />);
    wrapper
      .find('[data-cvent-id="quantityItems-item-quantityItem1"]')
      .find('[data-cvent-id="input"]')
      .hostNodes()
      .simulate('change', { target: { value: '10' } });
    await wait(0);
    expect(updateQuantity).toHaveBeenCalled();

    wrapper
      .find('[data-cvent-id="quantityItems-item-quantityItem1"]')
      .find('[data-cvent-id="input"]')
      .hostNodes()
      .simulate('change', { target: { value: '15' } });
    await wait(0);
    expect(updateLocalQuantity).toHaveBeenCalled();

    wrapper
      .find('[data-cvent-id="quantityItems-item-quantityItem1"]')
      .find('[data-cvent-id="input"]')
      .hostNodes()
      .simulate('change', { target: { value: '3' } });
    await wait(0);
    expect(updateLocalQuantity).toHaveBeenCalled();

    wrapper
      .find('[data-cvent-id="quantityItems-item-quantityItem1"]')
      .find('[data-cvent-id="input"]')
      .hostNodes()
      .simulate('change', { target: { value: '15' } });
    await wait(0);
    expect(updateLocalQuantity).toHaveBeenCalled();

    wrapper
      .find('[data-cvent-id="quantityItems-item-quantityItem1"]')
      .find('[data-cvent-id="input"]')
      .hostNodes()
      .simulate('blur');
    await wait(0);
    expect(updateQuantity).toHaveBeenCalled();
  });
});
