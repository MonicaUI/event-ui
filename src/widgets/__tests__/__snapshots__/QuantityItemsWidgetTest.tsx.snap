// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityItemsTest should render 1`] = `
<div
  className=""
  style={
    Object {
      "flex": "1 1 auto",
      "pointerEvents": undefined,
    }
  }
>
  <div
    data-cvent-id="quantityItems"
  >
    <h2
      className=""
      data-cvent-id="quantityItems-header"
      style={Object {}}
    >
      Header
    </h2>
    <div
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": "Instructional text",
        }
      }
      data-cvent-id="quantityItems-instructional-text"
      style={Object {}}
    />
    <ul
      className=""
      data-cvent-id="quantityItems-items"
      style={Object {}}
    >
      <li
        className=""
        style={Object {}}
      >
        <div
          className=""
          data-cvent-id="quantityItems-item-quantityItem1"
          style={Object {}}
        >
          <div
            className=""
            style={Object {}}
          >
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem1-name"
              style={Object {}}
            >
              Quantity Item Name
            </div>
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem1-code"
              style={Object {}}
            >
              Quantity Item Code
            </div>
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem1-description"
              style={Object {}}
            >
              Quantity Item Description
            </div>
          </div>
          <div
            className=""
            style={Object {}}
          >
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem1-fees"
              style={Object {}}
            >
              <span>
                50
              </span>
              <span
                className=""
                style={Object {}}
              >
                 EventWidgets_QuantityItems_EachQuantityItemLabelText__resx
              </span>
            </div>
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem1-select-dropdown"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="quantityItemSelectDropdown-quantityItem1"
                style={Object {}}
              >
                <span
                  className=""
                  style={Object {}}
                >
                  <span
                    data-cvent-id="label"
                  />
                </span>
              </label>
              <div
                className=""
                style={Object {}}
              >
                <select
                  className=""
                  data-cvent-id="input"
                  disabled={false}
                  id="quantityItemSelectDropdown-quantityItem1"
                  name="quantityItemSelectDropdown-quantityItem1"
                  onChange={[Function]}
                  required={false}
                  style={
                    Object {
                      "marginBottom": undefined,
                      "marginTop": undefined,
                      "paddingBottom": undefined,
                      "paddingTop": undefined,
                      "width": "100%",
                    }
                  }
                  value="0"
                >
                  <option
                    data-cvent-id="option-0"
                    value="0"
                  >
                    lbl_pop_0044
                  </option>
                  <option
                    data-cvent-id="option-1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    data-cvent-id="option-2"
                    value="2"
                  >
                    2
                  </option>
                  <option
                    data-cvent-id="option-3"
                    value="3"
                  >
                    3
                  </option>
                  <option
                    data-cvent-id="option-4"
                    value="4"
                  >
                    4
                  </option>
                  <option
                    data-cvent-id="option-5"
                    value="5"
                  >
                    5
                  </option>
                  <option
                    data-cvent-id="option-6"
                    value="6"
                  >
                    6
                  </option>
                  <option
                    data-cvent-id="option-7"
                    value="7"
                  >
                    7
                  </option>
                  <option
                    data-cvent-id="option-8"
                    value="8"
                  >
                    8
                  </option>
                  <option
                    data-cvent-id="option-9"
                    value="9"
                  >
                    9
                  </option>
                  <option
                    data-cvent-id="option-10"
                    value="10"
                  >
                    10+
                  </option>
                </select>
              </div>
            </div>
            <div>
              <div
                data-cvent-id="quantityItems-item-quantityItem1-capacity"
                style={Object {}}
              >
                EventWidgets_Products_CapacityRemaining__resx:{"numberRemaining":123}
              </div>
            </div>
          </div>
        </div>
      </li>
      <li
        className=""
        style={Object {}}
      >
        <div
          className=""
          data-cvent-id="quantityItems-item-quantityItemNoFee"
          style={Object {}}
        >
          <div
            className=""
            style={Object {}}
          >
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItemNoFee-name"
              style={Object {}}
            >
              Quantity Item No Fee
            </div>
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItemNoFee-code"
              style={Object {}}
            >
              quantity Item Code 2
            </div>
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItemNoFee-description"
              style={Object {}}
            >
              Quantity Item Description
            </div>
          </div>
          <div
            className=""
            style={Object {}}
          >
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItemNoFee-fees"
              style={Object {}}
            >
              <span>
                EventWidgets_Products_Free__resx
              </span>
            </div>
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItemNoFee-select-dropdown"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="quantityItemSelectDropdown-quantityItemNoFee"
                style={Object {}}
              >
                <span
                  className=""
                  style={Object {}}
                >
                  <span
                    data-cvent-id="label"
                  />
                </span>
              </label>
              <div
                className=""
                style={Object {}}
              >
                <select
                  className=""
                  data-cvent-id="input"
                  disabled={false}
                  id="quantityItemSelectDropdown-quantityItemNoFee"
                  name="quantityItemSelectDropdown-quantityItemNoFee"
                  onChange={[Function]}
                  required={false}
                  style={
                    Object {
                      "marginBottom": undefined,
                      "marginTop": undefined,
                      "paddingBottom": undefined,
                      "paddingTop": undefined,
                      "width": "100%",
                    }
                  }
                  value="0"
                >
                  <option
                    data-cvent-id="option-0"
                    value="0"
                  >
                    lbl_pop_0044
                  </option>
                  <option
                    data-cvent-id="option-1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    data-cvent-id="option-2"
                    value="2"
                  >
                    2
                  </option>
                  <option
                    data-cvent-id="option-3"
                    value="3"
                  >
                    3
                  </option>
                  <option
                    data-cvent-id="option-4"
                    value="4"
                  >
                    4
                  </option>
                  <option
                    data-cvent-id="option-5"
                    value="5"
                  >
                    5
                  </option>
                  <option
                    data-cvent-id="option-6"
                    value="6"
                  >
                    6
                  </option>
                  <option
                    data-cvent-id="option-7"
                    value="7"
                  >
                    7
                  </option>
                  <option
                    data-cvent-id="option-8"
                    value="8"
                  >
                    8
                  </option>
                  <option
                    data-cvent-id="option-9"
                    value="9"
                  >
                    9
                  </option>
                  <option
                    data-cvent-id="option-10"
                    value="10"
                  >
                    10+
                  </option>
                </select>
              </div>
            </div>
            <div>
              <div
                data-cvent-id="quantityItems-item-quantityItemNoFee-capacity"
                style={Object {}}
              >
                EventWidgets_Products_CapacityRemaining__resx:{"numberRemaining":123}
              </div>
            </div>
          </div>
        </div>
      </li>
      <li
        className=""
        style={Object {}}
      >
        <div
          className=""
          data-cvent-id="quantityItems-item-quantityItem2WithMaxPerInvitee"
          style={Object {}}
        >
          <div
            className=""
            style={Object {}}
          >
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem2WithMaxPerInvitee-name"
              style={Object {}}
            >
              Quantity Item 2 With Cap 1
            </div>
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem2WithMaxPerInvitee-code"
              style={Object {}}
            >
              Quantity Item 2 Code
            </div>
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem2WithMaxPerInvitee-description"
              style={Object {}}
            >
              Quantity Item 2 Description
            </div>
          </div>
          <div
            className=""
            style={Object {}}
          >
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem2WithMaxPerInvitee-fees"
              style={Object {}}
            >
              <span>
                50
              </span>
              <span
                className=""
                style={Object {}}
              >
                 EventWidgets_QuantityItems_EachQuantityItemLabelText__resx
              </span>
            </div>
            <div
              className=""
              data-cvent-id="quantityItems-item-quantityItem2WithMaxPerInvitee-select-dropdown"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="quantityItemSelectDropdown-quantityItem2WithMaxPerInvitee"
                style={Object {}}
              >
                <span
                  className=""
                  style={Object {}}
                >
                  <span
                    data-cvent-id="label"
                  />
                </span>
              </label>
              <div
                className=""
                style={Object {}}
              >
                <select
                  className=""
                  data-cvent-id="input"
                  disabled={false}
                  id="quantityItemSelectDropdown-quantityItem2WithMaxPerInvitee"
                  name="quantityItemSelectDropdown-quantityItem2WithMaxPerInvitee"
                  onChange={[Function]}
                  required={false}
                  style={
                    Object {
                      "marginBottom": undefined,
                      "marginTop": undefined,
                      "paddingBottom": undefined,
                      "paddingTop": undefined,
                      "width": "100%",
                    }
                  }
                  value="0"
                >
                  <option
                    data-cvent-id="option-0"
                    value="0"
                  >
                    lbl_pop_0044
                  </option>
                  <option
                    data-cvent-id="option-1"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    data-cvent-id="option-2"
                    value="2"
                  >
                    2
                  </option>
                </select>
              </div>
            </div>
            <div>
              <div
                data-cvent-id="quantityItems-item-quantityItem2WithMaxPerInvitee-capacity"
                style={Object {}}
              >
                EventWidgets_Products_CapacityRemaining__resx:{"numberRemaining":1}
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
